### PR TITLE
Improve the assertions for the downloaded solution metadata

### PR DESCRIPTION
**Pull request by kytrinyx**
The hard-coded JSON for the metadata was annoying me. The big JSON is hard to read. We improved it a bit by writing the JSON as multiline, and then compacting it using `json.Compact()` but it still felt brittle.

This improves it further by reading the JSON into a solution struct and just checking the fields we care about.

### CODE REVIEW

File: cmd/download_test.go [hunk 1]
```diff
@@ -113,7 +113,7 @@ func TestDownload(t *testing.T) {
 		defer os.RemoveAll(tmpDir)
 		assert.NoError(t, err)
 
-		ts := fakeDownloadServer(tc.requestor)
+		ts := fakeDownloadServer(fmt.Sprintf("%v", tc.requester))
```
Comment (nywilken) [inline cmd/download_test.go]: `strconv.FormatBool(tc.requester)` is what you want here.

File: cmd/download_test.go [hunk 2]
```diff
@@ -87,22 +87,22 @@ func TestDownload(t *testing.T) {
...
 			flags:       map[string]string{"exercise": "bogus-exercise"},
 		},
 		{
-			requestor:   requestorSelf,
+			requester:   true,
 			expectedDir: "",
 			flags:       map[string]string{"uuid": "bogus-id"},
 		},
 		{
-			requestor:   requestorOther,
```
Comment (nywilken) [inline cmd/download_test.go]: These constants are still defined below but no longer used

File: cmd/download_test.go (hunk 1 above)
Comment (kytrinyx) [reply cmd/download_test.go]: Ooooh, love it. Thanks!

File: cmd/download_test.go (hunk 2 above)
Comment (kytrinyx) [reply cmd/download_test.go]: Nice, thank you for catching that.

**Comment by kytrinyx** [comment]
I fixed the last details and force pushed, @nywilken.

**Pull request merged**